celery/utils/functional.py

```diff
@@ -334,8 +352,21 @@ def __reduce__(self):
     def __length_hint__(self):
         return self.__it.__length_hint__()
 
+    def __repr__(self):
+        if self.__done:
+            return repr(self.__consumed)
+        else:
+            return '[{}]'.format(', '.join(
```

Comment:
Please use `{0}` here, as position-less format is not supported by all 2.7 versions